src/Sulu/Bundle/SnippetBundle/Content/SingleSnippetSelection.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\SnippetBundle\Content;
+
+use Sulu\Bundle\SnippetBundle\Snippet\SnippetResolverInterface;
+use Sulu\Bundle\WebsiteBundle\ReferenceStore\ReferenceStoreInterface;
+use Sulu\Component\Content\Compat\PropertyInterface;
+use Sulu\Component\Content\Compat\Structure\PageBridge;
+use Sulu\Component\Content\PreResolvableContentTypeInterface;
+use Sulu\Component\Content\SimpleContentType;
+
+class SingleSnippetSelection extends SimpleContentType implements PreResolvableContentTypeInterface
+{
+    /**
+     * @var SnippetResolverInterface
+     */
+    private $snippetResolver;
+
+    /**
+     * @var ReferenceStoreInterface
+     */
+    private $snippetReferenceStore;
+
+    public function __construct(
+        SnippetResolverInterface $snippetResolver,
+        ReferenceStoreInterface $snippetReferenceStore
+    ) {
+        $this->snippetResolver = $snippetResolver;
+        $this->snippetReferenceStore = $snippetReferenceStore;
+
+        parent::__construct('SingleSnippetSelection', null);
+    }
+
+    public function getContentData(PropertyInterface $property)
+    {
+        $resolvedSnippet = $this->resolveSnippet($property);
```

Comment:
The single_snippet_selection should handle also the default snippets like the `snippet_selection` content type. Please have a look at the exist implementation of the Snippets Content Type to implement the same behaviour as there.

Reply:
are you sure? i dont think that feature is very helpful. especially because it is not implemented at all in the admin interface. also, if you want to implement that behaviour, you can just fallback to the default snippet in your template

Reply:
added the fallback mechanism. still not a fan of that feature 😇